common/scala/src/main/scala/whisk/connector/kafka/KafkaConsumerConnector.scala

```diff
@@ -69,13 +68,16 @@ class KafkaConsumerConnector(
         val thread = new Thread() {
             override def run() = {
                 while (!disconnect) {
-                    Try { peek() } match {
-                        case Success(records) =>
-                            records foreach { process.tupled(_) }
-                        case Failure(t) =>
-                            error(self, s"while polling: $t")
+                    Try {
+                        // Grab next batch of messages and commit offsets immediately
+                        // Process the message and commit afterwards. Otherwise the message would not be fetched again.
+                        // It won't be processed twice (tested in "KafkaConnectorTest")
+                        peek() foreach { process.tupled(_) }
+                        commit()
```

Comment:
comment says commit offsets immediately - but this is immediately. In fact what is different about the code path vs previous are the `recover` statements.

how does this not causes messages to be reprocessed if commit fails?